src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -202,5 +227,29 @@ internal static SafeCertContextHandle GetCertificateContext(X509Certificate cert
 
             return certContext;
         }
+
+        internal class ClearCertificateToRsaMap
+        {
+            ConcurrentDictionary<string, RSA> _rsaToCertificateMap;
```

Comment:
these references should be readonly

Reply:
@keegan-caruso This code pulled from a stale branch, disposing the RSA objects is fraught with issues.
Instead, just clear the dictionary.
If the number is reasonable, say 100 or 200, the memory pressure should be minimal.